core/src/components/cat-select/cat-select.tsx

```diff
@@ -0,0 +1,314 @@
+import { Component, Element, Event, EventEmitter, h, Host, Method, Prop } from '@stencil/core';
+import Choices, { Item } from 'choices.js';
+import { CatI18nRegistry } from '../cat-i18n/cat-i18n-registry';
+import { ClassNames, ItemFilterFn, ValueCompareFunction } from './interfaces';
+import { filterObject, isDefined } from './utils';
+import { Choice } from 'choices.js/src/scripts/interfaces/choice';
+import { Group } from 'choices.js/public/types/src/scripts/interfaces/group';
+
+export interface Option {
+  value: string;
+  label: string;
+  selected?: boolean;
+  disabled?: boolean;
+  choices?: Option[];
+}
+
+const getOptionTemplate = (data: Item): string => {
+  if (data.customProperties?.imageUrl) {
+    return `
+    <div class="d-flex align-items-center">
+      <img src="${data.customProperties.imageUrl}" style="margin-right: 0.5rem" />
+      ${data.label}
+    </div>
+    `;
+  }
+  return `<cat-checkbox label="${data.label}" checked="${data.selected}"></cat-checkbox>`;
+};
+
+@Component({
+  tag: 'cat-select',
+  styleUrl: 'cat-select.scss',
+  shadow: true
+})
+export class CatSelect {
+  private readonly i18n = CatI18nRegistry.getInstance();
+  private readonly searchResultLimit = 24;
+
+  private choice?: Choices | null = null;
+  private selectElement?: HTMLSelectElement;
+  private removeItemsButton?: HTMLCatButtonElement;
+  private choicesInner?: Element | null = null;
+
+  @Element() hostElement!: HTMLElement;
+
+  @Prop() multiple = false;
+  @Prop() position: 'auto' | 'top' | 'bottom' = 'auto';
+  @Prop() searchable = true;
+  @Prop() disabled = false;
+
+  // tags?
+  @Prop() addItems = true;
+  @Prop() editItems = false;
+  @Prop() paste = false;
+
+  @Prop() choices?: Option[];
```

Comment:
We still need to get the number of Props down to simplify the component. I will arrange a discussion meeting.